contracts/liquidator/OnChainLiquidator.sol

```diff
@@ -0,0 +1,535 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "./vendor/@uniswap/v3-core/contracts/interfaces/callback/IUniswapV3FlashCallback.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/base/PeripheryPayments.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/base/PeripheryImmutableState.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/PoolAddress.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/CallbackValidation.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol";
+import "./vendor/@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+
+import "../CometInterface.sol";
+import "../ERC20.sol";
+import "../IWstETH.sol";
+
+interface IUniswapV2Router {
+    function swapExactTokensForTokens(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external returns (uint256[] memory amounts);
+}
+
+// Balancer Interfaces
```

Comment:
Just out of interest, which of the assets we support are most liquid on Balancer?

Reply:
Balancer is good for swapping `wstETH` to `WETH`.

Curve is also good, but it only has a decent pool for `stETH` -> `ETH` so you have to unwrap `wstETH` and then wrap the `ETH` to `WETH`